src/vuex/modules/distribution.js

```diff
@@ -88,30 +87,31 @@ export default ({ node }) => {
         commit(`setDistributionError`, error)
       }
     },
-    async simulateWithdrawAllRewards({ rootState: { session }, dispatch }) {
+    async simulateWithdralRewards({ rootState: { session }, dispatch }) {
       return await dispatch(`simulateTx`, {
         type: `MsgWithdrawDelegationReward`,
         txArguments: {
-          toAddress: session.address,
-          validatorAddresses: []
+          toAddress: session.address
         }
       })
     },
-    async withdrawAllRewards(
+    async withdrawRewards(
       { rootState, getters, dispatch },
-      { gas, gasPrice, denom, validatorAddress, password, submitType }
+      { gas, gasPrice, denom, password, submitType }
     ) {
-      const top5Delegations = getTop5Delegations(getters.committedDelegations)
+      // Compares the amount in a [address1, {denom: amount}] array
+      const byDenom = denom => (a, b) => b[1][denom] - a[1][denom]
```

Comment:
very cool structure and comments! I like the chaining on line 105

Reply:
```suggestion
      const byBalanceOfDenom = denom => (a, b) => b[1][denom] - a[1][denom]
```

Reply:
Cheers. Much nicer than my attempt before.

Reply:
> very cool structure and comments! I like the chaining on line 105

Cheers. It's much nicer than my attempt before.